scrape_advanced.py

```
@@ -0,0 +1,261 @@
+    res.raise_for_status()
+
+    if write:
+        write_to_file(res, term)
+
+    return res.content
+
+
+def advanced_parse(content, db, term=''):
+    '''
+    Advanced parse takes the content from the request and then populates the database with the data
+    :param content: (html) The html containing the courses
+    :param db: (TinyDB) the current database
+    :return: None
+    '''
+    soup = BeautifulSoup(content, 'html5lib')
+
+    try:
+        table = soup.find('table', {'class': 'datadisplaytable'})
+        table_rows = table.find_all('tr')
+
+        table_headers = list()
+        start_idx = 0
+        for i, tr in enumerate(table_rows):
+            header_cols = tr.find_all('th', {'class': 'ddheader'})
+            for th in header_cols:
+                table_headers.append(get_parsed_text(th))
+            if table_headers:
+                start_idx = i
+                break
+
+        for tr in table_rows[start_idx:]:
+            try:
+                cols = tr.find_all('td', {'class': 'dddefault'})
+
+                if cols:
+                    s = defaultdict(lambda: defaultdict(list))
+
+                    num_blank = 0
+                    for i, c in enumerate(cols):
+                        a = c.find('a')
+                        cols[i] = get_parsed_text(a) if a else get_parsed_text(cols[i])
+                        if cols[i].isspace():
+                            num_blank += 1
+
+                    if num_blank > len(cols) - num_blank:
+                        raise BlankRow
+
+                    data = dict(zip(table_headers, cols))
+
+                    subject = data['Subj']
+                    key = data['Crse']
+                    crn = data['CRN']
+
+                    s[key][crn].append(data)
+
+                    j = dict(s)
+                    db.table(f'{subject}').insert(j)
+            except BlankRow:
+                continue
+    except AttributeError as e:
```

Catching all AttributeErrors produced in such a large are of code could easily swallow many errors that we do not want to be caught, and can be very hard to debug. Is there a way to avoid excepting these from such a large area, or entirely?